### ISSUE TITLE
--header not working for epinio app log

### ISSUE THREAD

**Issue by riqueps** [epinio/epinio#2560]
**Describe the bug**

When epinio server is protect with gcp iap solution, `epinio app logs` fails with `Invalid IAP credentials: empty token` error.
When IAP is disabled, the command works without any problem.

**Epinio CLI  output**
```
epinio app logs conda-header-test \
   --header "Proxy-Authorization: Bearer $(gcloud auth print-identity-token --include-email --audiences="AUDIENCE.apps.googleusercontent.com" --impersonate-service-account=SERVICE_ACCOUNT_NAME.iam.gserviceaccount.com)"  --trace-level 255
WARNING: This command is using service account impersonation. All API calls will be executed as [SERVICE_ACCOUNT_NAME.iam.gserviceaccount.com].
2023/09/08 10:47:20 Settings-0x140000d22d0: "level"=4 "msg"="Loading" "from"="/Users/henrique/Library/Application Support/epinio/settings.yaml"
2023/09/08 10:47:20 Settings-0x140000d22d0: "level"=4 "msg"="Loaded" "value"="namespace=(henrique), user=(), pass=(), access_token= 2023-09-08 18:42:39.095548 -0300 -03}), api=(https://epinio.DOMAIN.com), wss=(wss://epinio.DOMAIN.com), color=(true), appchart=(), @(/Users/henrique/Library/Application Support/epinio/settings.yaml)"
2023/09/08 10:47:20 EpinioClient/NewEpinioClient: "level"=4 "msg"="Ingress API" "url"="https://epinio.DOMAIN.com"
2023/09/08 10:47:20 EpinioClient/NewEpinioClient: "level"=4 "msg"="Settings API" "url"="https://epinio.DOMAIN.com"
2023/09/08 10:47:20 EpinioClient/Apps: "level"=4 "msg"="start" "Namespace"="henrique" "Application"="conda-header-test"
🚢  Streaming application logs
Namespace: henrique
Application: conda-header-test
2023/09/08 10:47:20 EpinioClient/Apps: "level"=5 "msg"="application logs" "Namespace"="henrique" "Application"="conda-header-test"
2023/09/08 10:47:20 EpinioApiClient: "level"=5 "msg"="executing request" "method"="GET" "uri"="" "body"="" "header"={"Proxy-Authorization":["Bearer "],"Authorization":["Bearer "]}
2023/09/08 10:47:21 EpinioApiClient: "level"=5 "msg"="response received" "status"=200 "body"="{\"token\":\""}" "header"={"Date":["Fri, 08 Sep 2023 13:47:21 GMT"],"Content-Length":["490"],"Via":["1.1 google"],"Alt-Svc":["h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000"],"Content-Type":["application/json; charset=utf-8"],"Epinio-Version":["v1.9.0"]} "TLSServerName"="epinio.DOMAIN.com" "status"=200
2023/09/08 10:47:21 EpinioApiClient: "level"=5 "msg"="response decoded" "response"={"token":""}
2023/09/08 10:47:21 EpinioApiClient: "msg"="decoding json error" "error"="invalid character 'I' looking for beginning of value" "body"="Invalid IAP credentials: empty token"
2023/09/08 10:47:21 EpinioClient/Apps: "level"=4 "msg"="return" "Namespace"="henrique" "Application"="conda-header-test"
❌  error streaming application logs: parsing error response: invalid character 'I' looking for beginning of value
```

**Expected behavior**
Retrieve epinio app logs

**To Reproduce**
Install epinio on GKE, expose it using an Application LoadBalance, enable IAP for epinio-server backend and execute: `epinio app logs APP_NAME --header "Proxy-Authorization: Bearer TOKEN"`

**Comment by thehejik** [comment]
Assigned myself to try and reproduce this with the IAP.

**Comment by thehejik** [comment]
@riqueps Hi, please is there an equivalent on AWS for IAP? I spent quite some time also with admin on GCP and I still don't have enough permissions and knowledge for enabling IAP there.

**Comment by riqueps** [comment]
@thehejik , I believe so, here is the [aws IaP documentation](https://docs.aws.amazon.com/prescriptive-guidance/latest/iap-on-aws/welcome.html), but I never configured it.

About GCP IAP, were you able to deploy epinio on gke and expose it via ALB?
If yes, follow the steps below:
- Setup Consent Screen - https://developers.google.com/workspace/guides/configure-oauth-consent
- Setup IAP
  - On Gcp console go to Security > identity Aware Proxy
  - You'll see your Backend Service, select the desired one and enable IAP (toggle button)
  - After enabled it, grant IAP-secured Web App User role to your user.
    - Select the Backend, on the left panel click on Add Principal

**Comment by thehejik** [comment]
@riqueps thanks, I passed this part already, the IAP works for me on `epinio-ui` (and probably on `-server` as well) but now I'm fighting with the `gcloud auth` command for generating the Bearer token, for some reason it resists and instead of returning a JWT token I see just errors.

Also it took me a while to figure out that I have to use "invisible" (from GKE PoV) `gce` ingress class  to make the IAP work.

Note: this helm command can be used for installing epinio with added the service/ingress annotations needed for IAP:
```bash 
helm upgrade --install epinio --namespace epinio epinio/epinio  --create-namespace \
  --set global.dex.enabled=false \
  --set global.domain=MY_PUBLIC_DOMAIN \
  --set ingress.annotations.'kubernetes\.io/ingress\.class=gce' \
  --set service.annotations.'cloud\.google\.com\/backend-config=\{"default": "config-default"\}' \
  --set epinioUI.service.annotations.'cloud\.google\.com\/backend-config=\{"default": "config-default"\}'
```

**Comment by riqueps** [comment]
Great @thehejik , yes, IAP protection can ben automated via GCE as well.
About epinio-server bearer token, there are two way to do.
Option 1- impersonate a service account
Option 2 - generate the token with your account (more complex)

**Option 1:**
- Create a SA and grant `Service Account Token Creator` and `AP-secured Web App` for it
- Grant `Service Account Token Creator` for your user
- Get the APP_AUDIENCE: this is the Client Id of the web Credential configured for IAP 
- Run epinio command, eg:
 ```
epinio login -u USER -p PWD https://EPINIO_SERVER_URL \
   --header "Proxy-Authorization: Bearer $(gcloud auth print-identity-token --include-email --audiences="APP_AUDIENCE.apps.googleusercontent.com" --impersonate-service-account=SERVICE_ACCOUNT_NAMEt@PROJECT_ID.iam.gserviceaccount.com)"
 ```

**Option 2:** https://cloud.google.com/iap/docs/authentication-howto#authenticating_from_a_desktop_app

**Comment by thehejik** [comment]
@riqueps Thanks, I finally managed to reproduce the issue and prepared the environment for our developers.

**Comment by gitricko** [comment]
@thehejik thanks for your effort!

**Comment by riqueps** [comment]
Great @thehejik, and tks for being so persistent on this!!

**Comment by thehejik** [comment]
Verified fixed in `v1.11.0-rc1` - tested on GKE with IAP enabled.
![image](https://github.com/epinio/epinio/assets/12828077/df80cc76-39ca-4072-8a6f-1f91987668d4)
